xarray/plot/dataset_plot.py

```diff
@@ -0,0 +1,393 @@
+                cmap_params, cbar_kwargs = _process_cmap_cbar_kwargs(
+                    plotfunc, locals(), ds[hue].values)
+
+            # subset that can be passed to scatter, hist2d
+            cmap_params_subset = dict(
+                (vv, cmap_params[vv])
+                for vv in ['vmin', 'vmax', 'norm', 'cmap'])
+
+        else:
+            cmap_params_subset = {}
+
+        primitive = plotfunc(ds=ds, x=x, y=y, hue=hue, hue_style=hue_style,
+                             ax=ax, cmap_params=cmap_params_subset, **kwargs)
+
+        if _is_facetgrid:  # if this was called from Facetgrid.map_dataset,
+            return primitive        # finish here. Else, make labels
+
+        if meta_data.get('xlabel', None):
+            ax.set_xlabel(meta_data.get('xlabel'))
+        if meta_data.get('ylabel', None):
+            ax.set_ylabel(meta_data.get('ylabel'))
+
+        if meta_data['add_legend']:
+            ax.legend(handles=primitive,
+                      labels=list(meta_data['hue_values']),
+                      title=meta_data.get('hue_label', None))
+        if meta_data['add_colorbar']:
+            cbar_kwargs = {} if cbar_kwargs is None else cbar_kwargs
+            if 'label' not in cbar_kwargs:
+                cbar_kwargs['label'] = meta_data.get('hue_label', None)
+            _add_colorbar(primitive, ax, cbar_ax, cbar_kwargs, cmap_params)
+
+        return primitive
+
+    @functools.wraps(newplotfunc)
+    def plotmethod(_PlotMethods_obj, x=None, y=None, hue=None,
+                   hue_style=None, col=None, row=None, ax=None,
+                   figsize=None,
+                   col_wrap=None, sharex=True, sharey=True, aspect=None,
+                   size=None, subplot_kws=None, add_guide=None,
+                   cbar_kwargs=None, cbar_ax=None, vmin=None, vmax=None,
+                   norm=None, infer_intervals=None, center=None, levels=None,
+                   robust=None, colors=None, extend=None, cmap=None,
+                   **kwargs):
+        """
+        The method should have the same signature as the function.
+
+        This just makes the method work on Plotmethods objects,
+        and passes all the other arguments straight through.
+        """
+        allargs = locals()
+        allargs['ds'] = _PlotMethods_obj._ds
+        allargs.update(kwargs)
+        for arg in ['_PlotMethods_obj', 'newplotfunc', 'kwargs']:
+            del allargs[arg]
+        return newplotfunc(**allargs)
+
+    # Add to class _PlotMethods
+    setattr(_Dataset_PlotMethods, plotmethod.__name__, plotmethod)
+
+    return newplotfunc
+
+
+@_dsplot
+def scatter(ds, x, y, ax, **kwargs):
+    """
+    Scatter Dataset data variables against each other.
+    """
+
+    if 'add_colorbar' in kwargs or 'add_legend' in kwargs:
+        raise ValueError('Dataset.plot.scatter does not accept '
+                         + '\'add_colorbar\' or \'add_legend\'. '
+                         + 'Use \'add_guide\' instead.')
```

Comment:
nit: it's a little more readable to use different quotation marks and implicit string concatenation, e.g.,
```
raise ValueError("this is the first line with a 'quote', "
                 "which continues here, too")
```